praw/models/reddit/base.py

```diff
@@ -18,7 +18,7 @@ def _url_parts(url):
             raise ClientException("Invalid URL: {}".format(url))
         return parsed.path.rstrip("/").split("/")
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: Union[Any, str]) -> bool:
```

Comment:
What benefit does `Union[Any, str]` offer over just `Any`? Isn't `str` a possible case for `Any`?

Reply:
It's more specific, so linters and IDEs can guess that strings are official. For example, Any could be anything, but if a string is specified then the IDE knows that it will work. If I gave it a dict object, it would break, but that's still under Any. If I gave it a str object, it will work 100% of the time, so that's why I added str

Reply:
Cool, thanks for the explanation.